pkg/registration/supply_chain_factory.go

```diff
@@ -48,53 +46,72 @@ func NewSupplyChainFactory(pType stitching.StitchingPropertyType) *SupplyChainFa
-	// Node supply chain builder
-	nodeSupplyChainNodeBuilder, err := f.buildNodeSupplyBuilder()
+	// Node supply chain template
+	nodeSupplyChainNode, err := f.buildNodeSupplyBuilder()
 	if err != nil {
 		return nil, err
 	}
+	glog.V(4).Infof("supply chain node : %++v", nodeSupplyChainNode)
 
-	// Pod supply chain builder
-	podSupplyChainNodeBuilder, err := f.buildPodSupplyBuilder()
+	// Resource Quota supply chain template
+	quotaSupplyChainNode, err := f.buildQuotaSupplyBuilder()
 	if err != nil {
 		return nil, err
 	}
+	glog.V(4).Infof("supply chain node : %++v", quotaSupplyChainNode)
 
-	// Resource Quota template
-	quotaSupplyChainNodeBuilder, err := f.buildQuotaSupplyBuilder()
+	// Pod supply chain template
+	podSupplyChainNode, err := f.buildPodSupplyBuilder()
 	if err != nil {
 		return nil, err
 	}
+	glog.V(4).Infof("supply chain node : %++v", podSupplyChainNode)
 
-	// Container suplly chain builder
-	containerSupplyChainNodeBuilder, err := f.buildContainer()
+	// Container supply chain template
+	containerSupplyChainNode, err := f.buildContainer()
 	if err != nil {
 		return nil, err
 	}
+	glog.V(4).Infof("supply chain node : %++v", containerSupplyChainNode)
 
-	// Application supply chain builder
-	appSupplyChainNodeBuilder, err := f.buildApplicationSupplyBuilder()
+	// Application supply chain template
+	appSupplyChainNode, err := f.buildApplicationSupplyBuilder()
 	if err != nil {
 		return nil, err
 	}
+	glog.V(4).Infof("supply chain node : %++v", appSupplyChainNode)
 
-	// Virtual application supply chain builder
-	vAppSupplyChainNodeBuilder, err := f.buildVirtualApplicationSupplyBuilder()
+	// Virtual application supply chain template
+	vAppSupplyChainNode, err := f.buildVirtualApplicationSupplyBuilder()
 	if err != nil {
 		return nil, err
 	}
+	glog.V(4).Infof("supply chain node : %++v", vAppSupplyChainNode)
 
 	supplyChainBuilder := supplychain.NewSupplyChainBuilder()
-	supplyChainBuilder.Top(vAppSupplyChainNodeBuilder)
-	supplyChainBuilder.Entity(appSupplyChainNodeBuilder)
-	supplyChainBuilder.Entity(containerSupplyChainNodeBuilder)
-	supplyChainBuilder.Entity(podSupplyChainNodeBuilder)
-	supplyChainBuilder.Entity(quotaSupplyChainNodeBuilder)
-	supplyChainBuilder.Entity(nodeSupplyChainNodeBuilder)
+	supplyChainBuilder.Top(vAppSupplyChainNode)
+	supplyChainBuilder.Entity(appSupplyChainNode)
+	supplyChainBuilder.Entity(containerSupplyChainNode)
+	supplyChainBuilder.Entity(podSupplyChainNode)
+	supplyChainBuilder.Entity(quotaSupplyChainNode)
+	supplyChainBuilder.Entity(nodeSupplyChainNode)
 
 	return supplyChainBuilder.Create()
 }
 
+func (f *SupplyChainFactory) buildNodeSupplyBuilder() (*proto.TemplateDTO, error) {
+	nodeSupplyChainNodeBuilder := supplychain.NewSupplyChainNodeBuilder(proto.EntityDTO_VIRTUAL_MACHINE)
+
+	nodeSupplyChainNodeBuilder = nodeSupplyChainNodeBuilder.
+		Sells(vCpuTemplateComm). // sells to Pods
+		Sells(vMemTemplateComm). // sells to Pods
+		// also sells VMPMAccess, Cluster to Pods
```

Comment:
the ClusterCommodity can be kept, it is always created during discovery.

Reply:
The issue is that the VM without pods are not considered for stitching, so the access commodities  from kubeturbo are not transferred to these VMs, which will cause errors when the kubeturbo node template is used for validation check.

Reply:
Nodes without Pods will not have VMPM_ACCESS commodity, but they still have CLUSTER commodity.  

Is CLUSTER commodity is also a kind of access commodity?

Reply:
yes

Reply:
Another question,  I thought the bug should be fixed from SDK or by changing the SE priority, but not by changing the supply chain definition in kubeturbo.

(1) Even node without pods are still can have VMPM_ACCESS commodities: 
[which are built from node labels](https://github.com/pallavidn/kubeturbo/blob/56c99d3204406143768ad41c5f155411c6f6e9be/pkg/discovery/dtofactory/node_entity_dto_builder.go#L147)).  The three nodes are all have labels. 

(2) The previous supplyChain definition for node does not have PMVM_ACCESS commodity.

Reply:
- The access commodities should not really cause supply chain validation errors.
- This fix is here is to match the kubeturbo VM node template  as close as to the VM template from VC, to ensure that any missing kubeturbo commodities will not cause validation errors.
- The root cause is that the priority of the VM templates for both the probes is default to 0 and  one of the template is randomly picked.
This is a workaround, we will need to discuss with the VC team to increase the priority of the VC probe VM template  so it is always picked and kubeturbo commodities are not used for validation checks.

Reply:
If the root cause is the priority or template type(Base Vs. Extension), then we can change the definition for the VirtualMachine node.
```go
/
type TemplateDTO struct {
	// The type of entity that the template represents. See EntityType
	// for the available types.
	TemplateClass *EntityDTO_EntityType `protobuf:"varint,1,req,name=templateClass,enum=common_dto.EntityDTO_EntityType" json:"templateClass,omitempty"`
	// The template type (Base or Extension), used during the validation process.
	TemplateType *TemplateDTO_TemplateType `protobuf:"varint,2,req,name=templateType,enum=common_dto.TemplateDTO_TemplateType" json:"templateType,omitempty"`
	// The priority of a Base template. For equivalent Base templates, Operations Manager uses the highest-priority
	// template, and discards discovered data from lower-priority Base templates.
	TemplatePriority *int32 `protobuf:"varint,3,req,name=templatePriority" json:"templatePriority,omitempty"`
	// This entity's list of {@link TemplateCommodity} items that it provides.
	CommoditySold []*TemplateCommodity `protobuf:"bytes,5,rep,name=commoditySold" json:"commoditySold,omitempty"`
	// The commodities bought from the different providers.
	// This Map contains the commodities bought where:
	CommodityBought []*TemplateDTO_CommBoughtProviderProp `protobuf:"bytes,6,rep,name=commodityBought" json:"commodityBought,omitempty"`
	// A map that defines the entity types that will be providers or consumers for this template entity.
	// The entry key is an entity type, from the EntityType enumeration. There can only be
	// one instance of each entity type in this map. The entry value is an instance of
	// ExternalEntityLink. Each entity link describes an entity type in the supply chain,
	// and the commodities it buys from or sells to the template entity.
	ExternalLink []*TemplateDTO_ExternalEntityLinkProp `protobuf:"bytes,7,rep,name=externalLink" json:"externalLink,omitempty"`
	// Each set represents a case where the entity must buy one commodity of the set ( a logical or of the set)
	// Note, the entity may buy more than one of the commodities in the set.
	CommBoughtOrSet  []*TemplateDTO_CommBoughtProviderOrSet `protobuf:"bytes,8,rep,name=commBoughtOrSet" json:"commBoughtOrSet,omitempty"`
	XXX_unrecognized []byte                                 `json:"-"`
}
```

Reply:
If the node without any pod won't enter into the stitching loop, then the node won't get stitched. The problem is still not solved?

Reply:
Will be opening a separate PR for this in the server.
We cannot use TEMPLATE.EXT if we want to run kubeturbo as standalone probe. Also, the default priority is 0, so it is the hypervisor probe that needs to increase the VM template priority to a much higher number. This also needs a PR and discussion on the server side.

Reply:
Ok. Let's issue two bugs about the VC probe and the reconciling processing, so that the 6.0.1 image can work without this patch.